### PR TITLE
Add openfga package

### LINKED ISSUE #1048: Add fga to Nix packages repository
**Summary**

Nix is a purely functional package manager and deployment system for POSIX ([example introduction](https://nixos.org/guides/nix-pills/why-you-should-give-it-a-try#why-you-should-give-it-a-try)).

That'd be great if `fga` could be added to the package registry: https://nixos.org/manual/nixpkgs/stable/#chap-quick-start

### PR DESCRIPTION
**Pull request by mjarosie**
## Description of changes

Add [openfga](https://github.com/openfga/openfga) to Nix packages repository. Will resolve https://github.com/openfga/openfga/issues/1048

It's my first time contributing to the nixpkgs repository, so I've added myself as a maintainer too. Let me know if I've missed anything. One question I have is - should I force the output of `go build` command to be in `./result/bin/` directory? Currently resulting bin ends up in `./result`.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/op/openfga/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildGoModule, fetchFromGitHub, go }:
+
+buildGoModule rec {
+  pname = "openfga";
+  version = "1.5.0";
+
+  src = fetchFromGitHub {
+    owner = "openfga";
+    repo = pname;
```
Comment (lolbinarycat) [inline pkgs/by-name/op/openfga/package.nix:9]:
```suggestion
    repo = "openfga";
```

using pname like this is discouraged.

File: pkgs/by-name/op/openfga/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    description = "A high performance and flexible authorization/permission engine built for developers and inspired by Google Zanzibar";
+    homepage = "https://openfga.dev/";
+    license = licenses.asl20;
+    mainProgram = "openfga";
+    maintainers = with maintainers; [ mjarosie ];
+  };
+
+  buildPhase = ''
+    go build -o $out/openfga ./cmd/openfga
+  '';
```
Comment (lolbinarycat) [inline pkgs/by-name/op/openfga/package.nix:26]:
```suggestion
  buildPhase = ''
    runHook preInstall
    go build -o $out/openfga ./cmd/openfga
    runHook postInstall
  '';
```

when overriding phases, you need to run their hooks manually.

**Review by lolbinarycat — COMMENTED** [review]
Hello, and welcome to nixpkgs!

Everything looks pretty good, there's just a few small issues i noticed.

File: pkgs/by-name/op/openfga/package.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchFromGitHub {
+    owner = "openfga";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-lqAoIOnizfvZm/NjGeinXzoci+ykmzdFtLDGeNoi97c=";
+  };
+
+  vendorHash = "sha256-vUfyo7/mapAzNs6RV+GVj5gOPEpMypNNIkoUgnBYTjA=";
+
```
Comment (sikmir) [inline pkgs/by-name/op/openfga/package.nix:15]:
```suggestion
  ldflags = [ "-s" "-w" "-X github.com/openfga/openfga/internal/build.Version=${version}" ];
```

File: pkgs/by-name/op/openfga/package.nix (hunk 2 above)
Comment (sikmir) [inline pkgs/by-name/op/openfga/package.nix:26]:
```suggestion
```

**Review by lolbinarycat — APPROVED** [review]

**Comment by mjarosie** [comment]
@lolbinarycat I just realised there were a couple of PRs opened over recent weeks for adding openfga (https://github.com/NixOS/nixpkgs/pull/295024, https://github.com/NixOS/nixpkgs/pull/297676) @lucperkins, @gaelreyrol I guess we should pick one?

**Comment by lucperkins** [comment]
I've closed my PR in favor of either this one or https://github.com/NixOS/nixpkgs/pull/297676. My vote is still for https://github.com/NixOS/nixpkgs/pull/297676 due its more comprehensive nature but will defer to others!

**Comment by mjarosie** [comment]
It seems there's been another PR merged in the meantime: https://github.com/NixOS/nixpkgs/pull/310534

**Comment by gepbird** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/310534

**Pull request closed without merging**